### PR TITLE
Deprecate /v2/sections/:section_id/students, update consumer

### PR DESCRIPTION
**Pull request by maddiedierker**
`GET /dashboardapi/sections/:section_id/students` already existed, so this PR: 
- updates the only consumer found (`PrintCertificates`) to point to dashboard instead of pegasus
- logs a notification to Honeybadger if this endpoint is hit

### CODE REVIEW

**Comment by Erin007** [comment]
My only concern is that `PrintCertificates` is used almost exclusively during Hour of Code. Is there any possibility that this route is used in other places, but "seasonally" so we'll miss them with this HB error check?

File: apps/src/templates/teacherDashboard/PrintCertificates.jsx [hunk 1]
```diff
@@ -38,7 +38,7 @@ class PrintCertificates extends Component {
   };
 
   onClickPrintCerts = () => {
-    $.ajax(`/v2/sections/${this.props.sectionId}/students`).done(result => {
+    $.ajax(`/dashboardapi/sections/${this.props.sectionId}/students`).done(result => {
       const names = result.map(student => student.name);
       this.setState({names}, this.submitForm);
```
Comment (Erin007) [inline apps/src/templates/teacherDashboard/PrintCertificates.jsx:43]: Any testing for `PrintCertificates` or anything that confirms this action still works as expected?

Comment (maddiedierker) [reply apps/src/templates/teacherDashboard/PrintCertificates.jsx:43]: currently there is [this unit test](https://github.com/code-dot-org/code-dot-org/blob/91beaac010797bafbe183417119d1a04ab6a8748/apps/test/unit/templates/teacherDashboard/PrintCertificatesTest.js#L35-L55), which mocks the return value for the component, so it doesn't test the endpoint itself. there are also [unit tests](https://github.com/code-dot-org/code-dot-org/blob/91beaac010797bafbe183417119d1a04ab6a8748/dashboard/test/controllers/api/v1/sections_students_controller_test.rb#L13-L37) on `GET /dashboardapi/sections/:section_id/students`, but no integration tests that i could find for `PrintCertificates`

Comment (davidsbailey) [reply apps/src/templates/teacherDashboard/PrintCertificates.jsx:43]: +1, please verify PrintCertificates some way before merging (not just in production after)

**Comment by maddiedierker** [comment]
@davidsbailey thanks for pointing out the documentation--i'm changing a lot of endpoints in that area, so i'll add a follow-up task to update that.

also, you mentioned that `ShowSecret` uses the old `/update` route, which will be gone once i merge #22739!

**Comment by davidsbailey** [comment]
@madelynkasula that's awesome, we're so close to getting rid of /v2/sections ! 🔥

**Comment by maddiedierker** [comment]
@Erin007 @davidsbailey i added a feature test to make sure `PrintCertificates` can still properly consume the new endpoint, so please take a look! 

basically, it checks that `/students` still returns something that can be mapped to an array of student names by `PrintCertificates`, which is then rendered in a textarea on `/certificates`

**Comment by davidsbailey** [comment]
sounds good, thanks for adding that!

**Pull request merged**